donate/payments/views.py

```diff
@@ -760,8 +761,23 @@ def get_initial(self):
         return initial
 
     def form_valid(self, form, send_data_to_basket=True):
-        if send_data_to_basket:
-            data = form.cleaned_data.copy()
+        data = form.cleaned_data.copy()
+
+        if hasattr(settings, 'POST_DONATE_NEWSLETTER_URL'):
```

Comment:
since it defaults to `None`, we need to check for that, something like:
```suggestion
        if hasattr(settings, 'POST_DONATE_NEWSLETTER_URL') and settings.POST_DONATE_NEWSLETTER_URL:
```

Reply:
actually, since we use django-environ to default it to None, we don'te really need a `hasattr` check right? it will either be a `String` or `None`

Reply:
updated the suggestion.